beacon-chain/attestation/service.go

```diff
@@ -92,12 +99,79 @@ func (a *Service) aggregateAttestations() {
 				continue
 			}
 			h := hashutil.Hash(enc)
-			if err := a.beaconDB.SaveAttestation(attestation); err != nil {
-				log.Errorf("Could not save attestation: %v", err)
+			// TODO(1366): Aggregate attestation with the existing one in cache. We need BLS.
+			log.Debugf("Forwarding aggregated attestation %#x to proposers through RPC", h)
+		}
+	}
+}
+
+// attestationPool takes an newly received attestation from sync service
+// and updates attestation pool.
+func (a *Service) attestationPool() {
+	incomingSub := a.incomingFeed.Subscribe(a.incomingChan)
```

Comment:
Having another routine subscribed to the same channel , ensures that any message only goes to 1 out of the 2 routines for any one message.
```go 
	go a.attestationAggregate()
	go a.attestationPool()
```
You can have one main goroutine which receives the attestation and then passes on the attestation to the two above goroutines

Reply:
I'm going to move `a.attestationAggregate` to operation service and move `a.attestationPool` to fork choice service in the next PR. We'll be ok here 👌